thorin/world.cpp

```diff
@@ -199,19 +200,6 @@ World::~World() {
     for (auto def : data_.defs_) def->~Def();
 }
 
-World World::stub() {
-    World w(name());
-    w.ostream_                 = ostream_;
-    w.state_                   = state_;
-    w.data_.imported_dialects_ = data_.imported_dialects_;
-
-    // bring dialects' axioms into new world.
-    Rewriter rewriter{w};
-    for (const auto& ax : data_.axioms_) rewriter.rewrite(ax.second);
```

Comment:
Is this removed without replacement?
I believe that is quite dangerous:
- `%mem.malloc` is parsed early on and added to the initial world, referencing types / params from the original world
- `%mem.alloc` is used in the actual program (reachable from a `.lam .extern`
- `%mem.alloc`->`%mem.malloc` is rewritten by `alloc2malloc`, or so it tries, but since the `%mem.malloc` still references the original world it is quite flakey

Reply:
This kind of flew under my radar and I was debugging for an hour or so before I realized that I removed these two crucial lines XD
Anyway, I have now added this directly to the Rewriter. If we make axioms external (which makes sense to me), we can remove this altogether.

Reply:
LGTM.
That should be alright, since `data_` is no longer automatically copied, so at least there are no broken axioms in the new world when using the `World(State &)` ctor 👍🏼

I agree that it'd be convenient to have axioms as externals for this case at least.
Not sure I'd be happy with it though, the name external is kind of misleading. I understand the external things to be accessible externally, e.g. via FFI calls (from C), which is not the case for axioms.